### PR TITLE
fix: always cast PartialMessage(able) IDs to int

### PR DESCRIPTION
**Pull request by NeloBlivion**
## Summary

This was initially a PR to fix broken views sent in create_thread, but it turned out to be an underlying issue... once again with string IDs still being present in some areas of the library. `ForumChannel.create_thread` used the raw string ID for the ViewStore key, resulting in it always failing to resolve. Instead of casting `int` here, I opted to force `PartialMessage` and `PartialMessageable` to always convert to `int`.

## Information

- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.
- [x] I have updated the changelog to include these changes.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -28,6 +28,10 @@ These changes are available on the `master` branch, but have not yet been releas
   ([#2390](https://github.com/Pycord-Development/pycord/pull/2390))
 - Fixed `NameError` in some instances of `Interaction`.
   ([#2402](https://github.com/Pycord-Development/pycord/pull/2402))
+- Fixed `PartialMessage/able` IDs sometimes being `str` rather than `int`.
+  ([#2406](https://github.com/Pycord-Development/pycord/pull/2406))
+- Fixed Views sent in `ForumChannel.create_thread` not recieving interactions.
+  ([#2406](https://github.com/Pycord-Development/pycord/pull/2406))
```
Comment (Lulalaby) [inline CHANGELOG.md:34]:
```suggestion
```
Why is that in here

**Review by Lulalaby — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (Lulalaby) [reply CHANGELOG.md:34]: I feel like it's duplicate

Comment (NeloBlivion) [reply CHANGELOG.md:34]: This was the main issue i found which led to me making this fix, but if it's not necessary then fair enough

Comment (Lulalaby) [reply CHANGELOG.md:34]: I mean the real change is the first entry. Maybe combine it?

Comment (Dorukyum) [inline CHANGELOG.md:34]:
```suggestion
- Fixed interactions being ignored due to `PartialMessage(able).id` being of type `str`.
  ([#2406](https://github.com/Pycord-Development/pycord/pull/2406))
```

File: discord/channel.py [hunk 2]
```diff
@@ -3189,7 +3189,7 @@ def __init__(
     ):
         self._state: ConnectionState = state
         self._channel: Object = Object(id=id)
-        self.id: int = id
+        self.id: int = int(id)
```
Comment (Dorukyum) [inline discord/channel.py:3192]: Why not change the part where the ID is passed in with the wrong type?

**Review by Dorukyum — CHANGES_REQUESTED** [review]

File: discord/channel.py (hunk 2 above)
Comment (NeloBlivion) [reply discord/channel.py:3192]: I suppose, but shouldn't we guarantee all IDs are int for parity with full channel objects?

Comment (Dorukyum) [reply discord/channel.py:3192]: The library usually converts before passing in, so I think we should continue that

**Comment by Dorukyum** [comment]
I *think* I fixed what you said was the root issue, don't have the chance to test right now.

**Comment by NeloBlivion** [comment]
That is the correct fix, yes

**Review by Dorukyum — APPROVED** [review]

**Review by Lulalaby — APPROVED** [review]

**Pull request merged**